optaplanner-examples/src/test/java/org/optaplanner/examples/common/app/SolverPerformanceTest.java

```
@@ -94,65 +95,60 @@ public void setUp() {
+        assertThat(bestScore)
                 .as("The bestScore (" + bestScore + ") must be at least the bestScoreLimit (" + bestScoreLimit + ").")
-                .isGreaterThanOrEqualTo(0);
-
-        try (InnerScoreDirector<Solution_, Score_> scoreDirector = scoreDirectorFactory.buildScoreDirector()) {
-            scoreDirector.setWorkingSolution(bestSolution);
-            Score_ score = scoreDirector.calculateScore();
-            assertThat(bestScore).isEqualTo(score);
-            if (scoreDirector.isConstraintMatchEnabled()) {
-                Map<String, ConstraintMatchTotal<Score_>> constraintMatchTotals =
-                        scoreDirector.getConstraintMatchTotalMap();
-                assertThat(constraintMatchTotals).isNotNull();
-                assertThat(constraintMatchTotals.values().stream()
-                        .map(ConstraintMatchTotal::getScore)
-                        .reduce(Score::add)
-                        .orElse(scoreDefinition.getZeroScore())).isEqualTo(score);
-                assertThat(scoreDirector.getIndictmentMap()).isNotNull();
-            }
+                .isGreaterThanOrEqualTo(bestScoreLimit);
+
+        try {
+            ScoreExplanation<Solution_, Score_> scoreExplanation = scoreManager.explainScore(bestSolution);
+            Map<String, ConstraintMatchTotal<Score_>> constraintMatchTotals =
+                    scoreExplanation.getConstraintMatchTotalMap();
+            assertThat(constraintMatchTotals).isNotNull();
+            assertThat(constraintMatchTotals.values().stream()
+                    .map(ConstraintMatchTotal::getScore)
+                    .reduce(Score::add)
+                    .orElse(bestScore.zero()))
+                            .isEqualTo(scoreExplanation.getScore());
+            assertThat(scoreExplanation.getIndictmentMap()).isNotNull();
+        } catch (IllegalStateException ex) {
+            logger.info("Constraint match likely not supported.", ex);
```

important

This eats an exception! Let's not do that.